File: src/JavaScript/easy/array/problem217.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */


var containsDuplicate = function(nums) {
    let numsObj = {}
    for(let i = 0; i < nums.length; i++){
        if(numsObj?.[nums[i]]){
            return true
        }  else {
            numsObj[nums[i]] = 1
        }
    }
    return false
};


let nums = [1,2,3,1]
// let nums = [1,2,3,4]

containsDuplicate(nums)

File: src/Python/easy/array/problem217.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        numsObj = {}
        for num in nums:
            if num in numsObj:
                return True
            else:
                numsObj[num] = 1
        return False
    

nums = [1,2,3,1]


solution = Solution()
solution.containsDuplicate(nums)